src/dualize.jl

```diff
@@ -120,4 +119,33 @@ function get_objective_coefficients(model::MOI.ModelLike)
 
     b0 = model.objective.constant # Constant term of the objective function
     return a0, b0
+end
+
+"""
+        add_dualmodel_equality_constraints(dualmodel::MOI.ModelLike, model::MOI.ModelLike, dict_constr_coeffs::Dict, 
+                                            dict_dualvar_primalcon::Dict, a0::Array{T, 1}) where T
+
+Add the dual model equality constraints
+"""
+function add_dualmodel_equality_constraints(dualmodel::MOI.ModelLike, model::MOI.ModelLike, dict_constr_coeffs::Dict, 
+                                            dict_dualvar_primalcon::Dict, a0::Array{T, 1}) where T
+    
+    sense = MOI.get(dualmodel, MOI.ObjectiveSense()) # Get dual model sense
+
+    for var = 1:model.num_variables_created #Number of variables
+        safs = Array{MOI.ScalarAffineTerm{T}}(undef, model.nextconstraintid) 
+        for constr = 1:model.nextconstraintid # Number of constraints of the model
+            vi = VI(constr)
+            term = dict_constr_coeffs[dict_dualvar_primalcon[vi]][1][var] # Accessing Ai^T
+            safs[constr] = MOI.ScalarAffineTerm(term, vi)
+            
+        end
+        # Add constraint, the sense of a0 depends on the dualmodel ObjectiveSense
+        if sense == MOI.MAX_SENSE 
+            MOI.add_constraint(dualmodel, MOI.ScalarAffineFunction(safs, a0[var]), MOI.EqualTo(0.0))
+        else
+            MOI.add_constraint(dualmodel, MOI.ScalarAffineFunction(safs, -a0[var]), MOI.EqualTo(0.0))
```

Comment:
the only difference with the above line is a `-`. it would be preferable to minimize code duplication and perhaps do something like `...safs, (sense == MOI.MAX_SENSE ? 1 : -1) * a0[var]), ...`

Reply:
Very good tip! :+1: